pkg/controllers/clusterinfo/clusterinfo_controller.go

```diff
@@ -188,15 +188,21 @@ func (r *ClusterInfReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 
 	// TODO: the conditions of managed cluster need to be deprecated.
-	newConditions := cluster.Status.Conditions
+	clusterConditions := cluster.Status.Conditions
+	newClusterInfo := clusterInfo.DeepCopy()
+
+	// empty clusterInfo status when cluster is offline
+	if utils.ClusterIsOffLine(clusterConditions) {
+		newClusterInfo.Status = clusterinfov1beta1.ClusterInfoStatus{}
```

Comment:
I think we only empty the node list here.

Reply:
done